signalfx-java/src/main/java/com/signalfx/metrics/connection/AbstractHttpDataPointProtobufReceiverConnection.java

```diff
@@ -55,6 +55,7 @@ public void addDataPoints(String auth, List<SignalFxProtocolBuffers.DataPoint> d
                 }
             } finally {
                 if (resp != null) {
+                    resp.getEntity().getContent().close();
```

Comment:
Are you fairly certain that `getEntity()` will always return non-null?  According to [docs](https://hc.apache.org/httpcomponents-core-ga/httpcore/apidocs/org/apache/http/HttpResponse.html#getEntity()) it can return null (looks like `getContent()` is safe though). E.g. maybe if there is a proxy that returns no body for whatever reason (our own ingest server/gateway would always return a body but no guarantees).

Reply:
@keitwb Thanks for checking it! I did not do null check previously under the assumption that the ingest endpoint will always return body, and the null check has been added in the latest commit